gateways/orbital/types.go

```diff
@@ -0,0 +1,127 @@
+package orbital
+
+import "encoding/xml"
+
+type RequestType string
+
+const (
+	RequestTypeAuth    = "NewOrder"
+	RequestTypeCapture = "MarkForCapture"
+	RequestTypeVoid    = "Reversal"
+	RequestTypeRefund  = "Reversal"
+)
+
+type MessageType string
+
+const (
+	MessageTypeAuth           MessageType = "A"
+	MessageTypeAuthAndCapture MessageType = "AC"
+	MessageTypeCapture        MessageType = "FC"
+	MessageTypeRefund         MessageType = "R"
+)
+const TerminalIDStratus string = "001"
+
+type BIN string
+
+const (
+	BINStratus BIN = "000001"
+	BINPNS     BIN = "000002"
+)
+
+type IndustryType string
+
+const (
+	IndustryTypeEcomm       IndustryType = "EC"
+	IndustryTypeInstallment IndustryType = "IN"
+	IndustryTypeIVR         IndustryType = "IV"
+	IndustryTypeMailOrder   IndustryType = "MO"
+	IndustryTypeRecurring   IndustryType = "RC"
+)
+
+type CardSecValInd int
+
+const (
+	CardSecPresent      CardSecValInd = 1
+	CardSecIllegible    CardSecValInd = 2
+	CardSecNotAvailable CardSecValInd = 9
+)
+
+type CVVResponseCode string
+
+const (
+	CVVResponseMatched      CVVResponseCode = "M"
+	CVVResponseNotMatched   CVVResponseCode = "N"
+	CVVResponseNotProcessed CVVResponseCode = "P"
+	CVVResponseNotPresent   CVVResponseCode = "S"
+	CVVResponseUnsupported  CVVResponseCode = "U"
+	CVVResponseNotValidI    CVVResponseCode = "I"
+	CVVResponseNotValidY    CVVResponseCode = "Y"
+)
+
+type AVSResponseCode string
+
+const (
+	AVSResponseNotChecked             AVSResponseCode = "3"
+	AVSResponseSkipped4               AVSResponseCode = "4"
+	AVSResponseSkippedR               AVSResponseCode = "R"
+	AVSResponseMatch                  AVSResponseCode = "H"
+	AVSResponseNoMatch                AVSResponseCode = "G"
+	AVSResponseZipMatchAddressNoMatch AVSResponseCode = "A"
+	AVSResponseZipNoMatchAddressMatch AVSResponseCode = "F"
+)
+
+type Request struct {
```

Comment:
these seem a little odd to me, why XML Name?

Reply:
All requests can use the same structure and properties but the name of the actual tag when marshaled to XML has to change, xmlName lets you change it. EG :

an auth request looks like:
```
<Request> //This is the Request Struct
  <NewOrder>...</NewOrder> //this is a RequestBody struct with xmlName = "NewOrder"
</Request>
```

a void request looks like:
```
<Request> //This is the Request Struct
  <Reversal>...</Reversal> //this is a RequestBody struct with xmlName = "Reversal"
</Request>
```

Setting it looks like this :
`orbitalAuthRequest.Body.XMLName = xml.Name{Local: RequestTypeAuth}`

Reply:
Thanks got it